File: config-api/src/main/java/com/config/api/ia_backend/service/PreviewService.java
package com.config.api.ia_backend.service;

import com.config.api.ia_backend.client.PredictApiClient;
import com.config.api.ia_backend.dto.PreviewRequest;
import com.config.api.ia_backend.dto.PreviewResponse;
import org.springframework.stereotype.Service;

@Service
public class PreviewService {
    private final PredictApiClient client;
    public PreviewService(PredictApiClient client){ this.client = client; }
    public PreviewResponse preview(PreviewRequest req){ return client.previewIntent(req); }
}

File: banca-ia-front/src/app/services/preview.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { API_BASE } from '../core/api.config';
import { Observable } from 'rxjs';
import { PreviewRequest, PreviewResponse } from '../models/predict.model';

@Injectable({ providedIn: 'root' })
export class PreviewService {
  constructor(private http: HttpClient) {}
  intentPreview(body: PreviewRequest): Observable<PreviewResponse> {
    return this.http.post<PreviewResponse>(`${API_BASE}/preview/intent`, body);
  }
}
